### DISCUSSION TITLE
How are you versioning your cue files?

### DISCUSSION THREAD

**Discussion by jugaadi** [cuelang/cue#960] [Q&A] [answered]
1. Do you use semantic versioning?
2. Can we automate cue file versioning i.e generate a version based on the changes?

**Comment by mpvl** [comment, marked as answer]
Currently there is no versioning mechanism (i.e. users would have to do so themselves). There is a proposal under way related to versioning CUE modules, at least, that goes in this direction.

**Reply by jugaadi** [reply]
Thanks @mpvl  . Any best practices?

**Reply by myitcv** [reply]
Best practices now involved simply vendoring your dependencies in `cue.mod/pkg`. Best practices for module versioning are being worked on as part of the proposal that @mpvl referenced.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/960.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
